test/tstDetailsAlgorithms.cpp

```diff
@@ -157,6 +159,19 @@ BOOST_AUTO_TEST_CASE(intersects)
   BOOST_TEST(intersects(Point{-1., 0., 0.}, sphere));
   BOOST_TEST(intersects(Point{-0.6, -0.8, 0.}, sphere));
   BOOST_TEST(!intersects(Point{-0.7, -0.8, 0.}, sphere));
+
+  // triangle
+  using Point2 = ArborX::ExperimentalHyperGeometry::Point<2>;
+  constexpr ArborX::ExperimentalHyperGeometry::Triangle<2> triangle{
+      {{0, 0}}, {{1, 0}}, {{0, 2}}};
+  BOOST_TEST(intersects(Point2{{0, 0}}, triangle));
+  BOOST_TEST(intersects(Point2{{1, 0}}, triangle));
```

Comment:
Please test with the 3rd support point too.

Reply:
What about points on the edges like (.5, 0.)?

Reply:
> What about points on the edges like (.5, 0.)?

(0.5, 1) is on the edge.

Reply:
maybe we should have center of all 3 edges?

Reply:
Sure.